generator/ServiceClientGeneratorLib/Generators/SourceFiles/ServiceClientsNetStandard.tt

```diff
@@ -483,7 +483,7 @@ namespace <#=this.Config.Namespace#>
             return InvokeAsync<<#=operation.Name#>Response>(request, options, cancellationToken);
         }
 <#
-        if (this.Config.ServiceModel.H2Support == H2SupportDegree.EventStream && operation.IsEventStreamOutput)
+        if ((this.Config.ServiceModel.H2Support == H2SupportDegree.Optional && operation.IsEventStreamBidi) || (this.Config.ServiceModel.H2Support == H2SupportDegree.EventStream && operation.IsEventStreamOutput))
```

Comment:
Problem with my original logic as well but we should be checking if `H2Support` is set to `Required` and if so add the `#if NET8_0_OR_GREATER`.